src/layer/roialign.cpp

```diff
@@ -20,6 +20,117 @@ namespace ncnn {
 
 DEFINE_LAYER_CREATOR(ROIAlign)
 
+// adapted from detectron2
+// https://github.com/facebookresearch/detectron2/blob/master/detectron2/layers/csrc/ROIAlign/ROIAlign_cpu.cpp
+template <typename T>
+struct PreCalc {
+  int pos1;
+  int pos2;
+  int pos3;
+  int pos4;
+  T w1;
+  T w2;
+  T w3;
+  T w4;
+};
+
+template <typename T>
+void pre_calc_for_bilinear_interpolate(
```

Comment:
The accelerated version shall be put into x86/arm/... subclass implementation
Keep src/layer/xyz.cpp simple and naive, and helpful for autotest

Reply:
solved : )